lib/kernel-manager.coffee

```diff
@@ -149,9 +151,9 @@ module.exports = KernelManager =
 
         kernelSpec.grammarLanguage = grammarLanguage
 
-        ConfigManager.writeConfigFile (filepath, config, onlyConnect = false) =>
-            kernel = new Kernel kernelSpec, grammar, config, filepath, onlyConnect
+        customKernelConnectionPath = path.join atom.project.rootDirectories[0].path, 'hydrogen', 'connection.json'
 
+        finishKernelStartup = (kernel) =>
```

Comment:
@tiangolo Oh! I missed that `finishKernelStartup` was called in the callback to `ConfigManager.writeConfigFile`. Now I understand the need for `finishKernelStartup`.